### PR TITLE
fix: show reason for request denial in the error

### PR DESCRIPTION
**Pull request by khoaguin**
Issue: [Cannot access the denial reason in syft without using notifications (should be added in the syfterror that says it was denied)](https://github.com/OpenMined/Heartbeat/issues/757)

closes https://github.com/OpenMined/Heartbeat/issues/757

### CODE REVIEW

File: packages/syft/src/syft/external/oblv/oblv_service.py [hunk 1]
```diff
@@ -376,9 +376,12 @@ def send_user_code_inputs_to_enclave(
                 message=f"Unable to find {user_code_id} in {type(user_code_service)}"
             )
         user_code = user_code.ok()
-
+        if "reason" in context.extra_kwargs.keys():
```
Comment (shubham3121) [inline packages/syft/src/syft/external/oblv/oblv_service.py]: we can replace this with 
```reason = context.extra_kwargs.get("reason", "")```, where default can empty string.

File: packages/syft/src/syft/service/code/user_code.py [hunk 2]
```diff
@@ -795,7 +810,7 @@ def add_custom_status(context: TransformContext) -> TransformContext:
             verify_key=context.node.signing_key.verify_key,
         )
         context.output["status"] = UserCodeStatusCollection(
-            status_dict={node_identity: UserCodeStatus.PENDING}
+            status_dict={node_identity: (UserCodeStatus.PENDING, "No reason")}
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py]: Lets set the default reason string to empty string.

File: packages/syft/src/syft/service/code/user_code.py [hunk 3]
```diff
@@ -804,7 +819,7 @@ def add_custom_status(context: TransformContext) -> TransformContext:
         # else:
         #     raise ValueError(f"Invalid input keys: {input_keys} for {node_identity}")
     elif context.node.node_type == NodeType.ENCLAVE:
-        status_dict = {key: UserCodeStatus.PENDING for key in input_keys}
+        status_dict = {key: (UserCodeStatus.PENDING, "No reason") for key in input_keys}
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py]: Same here, 

```(UserCodeStatus.PENDING, "")```

File: packages/syft/src/syft/service/enclave/enclave_service.py [hunk 4]
```diff
@@ -57,8 +57,12 @@ def send_user_code_inputs_to_enclave(
         if isinstance(user_code, SyftError):
             return user_code
 
+        if "reason" in context.extra_kwargs.keys():
+            reason = context.extra_kwargs["reason"]
```
Comment (shubham3121) [inline packages/syft/src/syft/service/enclave/enclave_service.py]: Similarly we can replace this with:
reason = context.extra_kwargs.get("reason", ""), where default can empty string.

File: packages/syft/src/syft/service/request/request.py [hunk 5]
```diff
@@ -807,16 +807,20 @@ def valid(self) -> Union[SyftSuccess, SyftError]:
         return SyftSuccess(message=f"{type(self)} valid")
 
     def mutate(self, obj: UserCode, context: ChangeContext, undo: bool) -> Any:
+        if "reason" in context.extra_kwargs.keys():
+            reason = context.extra_kwargs["reason"]
```
Comment (shubham3121) [inline packages/syft/src/syft/service/request/request.py]: Same here:
we can replace this with
reason = context.extra_kwargs.get("reason", ""), where default can empty string.

**Review by shubham3121 — CHANGES_REQUESTED** [review]
Left some minor comments, otherwise looks good.

Nice work @khoaguin !!

Can you also add a screenshot of the change.

**Comment by khoaguin** [comment]
screen shots are in the ticket https://github.com/orgs/OpenMined/projects/81/views/1?pane=issue&itemId=39964760

**Review by shubham3121 — APPROVED** [review]

**Pull request merged**